src/VisualStudio/Core/Def/Implementation/Experimentation/AnalyzerVsixSuggestedActionCallback.cs

```diff
@@ -61,7 +63,7 @@ public void OnSuggestedActionExecuted(SuggestedAction action)
                 return;
             }
 
-            EnsureInitialized();
+            ThreadingContext.JoinableTaskFactory.Run(() => EnsureInitializedAsync(CancellationToken.None));
```

Comment:
if we are going to do this anyway.. not sure why it don't do it inside of IExperiementService itself rather than asking all caller to do this.

we can basically have IExpereiemtnService.IsEnabledAsync and ISEnabled basically doing what it is doing here
JTF.Run(() => IsEnabledAsync(...));

Reply:
This is never the desirable solution. We always want to push incorrect asynchrony as far to the callers as possible.